src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -546,19 +548,41 @@ private function computeDoc(Documentation $documentation, \ArrayObject $definiti
             'paths' => $paths,
         ];
 
+        $securityDefinitions = [];
+        $security = [];
+
         if ($this->oauthEnabled) {
-            $doc['securityDefinitions'] = [
-                'oauth' => [
-                    'type' => $this->oauthType,
-                    'description' => 'OAuth client_credentials Grant',
-                    'flow' => $this->oauthFlow,
-                    'tokenUrl' => $this->oauthTokenUrl,
-                    'authorizationUrl' => $this->oauthAuthorizationUrl,
-                    'scopes' => $this->oauthScopes,
-                ],
+            $securityDefinitions['oauth'] = [
+                'type' => $this->oauthType,
+                'description' => 'OAuth client_credentials Grant',
+                'flow' => $this->oauthFlow,
+                'tokenUrl' => $this->oauthTokenUrl,
+                'authorizationUrl' => $this->oauthAuthorizationUrl,
+                'scopes' => $this->oauthScopes,
             ];
 
-            $doc['security'] = [['oauth' => []]];
+            $security[] = ['oauth' => []];
+        }
+
+        if ($this->apiKeys) {
+            foreach ($this->apiKeys as $key => $apiKey) {
+                $name = $apiKey['name'];
+                $type = $apiKey['type'];
+
+                $securityDefinitions[$key] = [
+                    'type' => 'apiKey',
+                    'in' => $type,
+                    'description' => sprintf('Value for the %s %s', $name, $type === 'query' ? sprintf('%s parameter', $type) : $type),
+                    'name' => $name,
+                ];
+
+                $security[] = [$key => []];
```

Comment:
This only allows a logical OR but AND support can be added in another PR.

e.g.:
```
// OR
$security = [];
$security[] = ['foo' => []];
$security[] = ['bar' => []];

// AND
$security = [];
$security[0] = ['foo' => []];
$security[0] += ['bar' => []];
```